MinecraftClient/ChatBots/AutoEat.cs

```diff
@@ -20,15 +21,22 @@ public AutoEat(int Threshold)
 
         public override void OnHealthUpdate(float health, int food)
         {
-            if (food <= HungerThreshold || (food < 20 && health < 20))
+            if (((food <= HungerThreshold) || (food < 20 && health < 20)) && !Eating)
             {
-                Eating = true;
-                FindFoodAndEat();
+                Eating = FindFoodAndEat();
+                if (!Eating)
+                    ChangeSlot(LastSlot);
             }
             // keep eating until full
             if (food < 20 && Eating)
             {
-                FindFoodAndEat();
+                Task.Factory.StartNew(delegate
```

Comment:
Unfortunately, MCC is not fully thread safe and might cause an exception when using the ChatBot API from a different thread. As a workaround, you can set a boolean like `EatOnNextUpdate = true` (or use a counter decremented each tick), then do the logic in `OnUpdate()`. Apologies for the low code quality of MCC's core 😅

Reply:
Don’t mind. I have came up with a better solution that do not require Threading. I will modify and test the code tomorrow.